### ISSUE TITLE
Dispatch calls should return actions

### ISSUE THREAD

**Issue by vitorprado** [Reduks/Reduks#7]
Actually the return of dispatch calls is state, this is wrong because the state should be accessed via subscriber and with action returns we can add some middlewares (like Rx and Promisses).

**Comment by bloderxd** [comment]
Could you give more information about it? Dispatch calls has not returning as you can see

```kotlin
fun dispatch(action: Action<State>) {
        reducer(state, action)
}
```

Are you saying about all dispatch action, like call reducers?

**Comment by vitorprado** [comment]
If you look into the redux JS implementation you'll find this:

```Javascript
/**
   * Dispatches an action. It is the only way to trigger a state change.
   *
   * The `reducer` function, used to create the store, will be called with the
   * current state tree and the given `action`. Its return value will
   * be considered the **next** state of the tree, and the change listeners
   * will be notified.
   *
   * The base implementation only supports plain object actions. If you want to
   * dispatch a Promise, an Observable, a thunk, or something else, you need to
   * wrap your store creating function into the corresponding middleware. For
   * example, see the documentation for the `redux-thunk` package. Even the
   * middleware will eventually dispatch plain object actions using this method.
   *
   * @param {Object} action A plain object representing “what changed”. It is
   * a good idea to keep actions serializable so you can record and replay user
   * sessions, or use the time travelling `redux-devtools`. An action must have
   * a `type` property which may not be `undefined`. It is a good idea to use
   * string constants for action types.
   *
   * @returns {Object} For convenience, the same action object you dispatched.
   *
   * Note that, if you use a custom middleware, it may wrap `dispatch()` to
   * return something else (for example, a Promise you can await).
   */
  function dispatch(action) {
    if (!isPlainObject(action)) {
      throw new Error(
        'Actions must be plain objects. ' +
          'Use custom middleware for async actions.'
      )
    }

    if (typeof action.type === 'undefined') {
      throw new Error(
        'Actions may not have an undefined "type" property. ' +
          'Have you misspelled a constant?'
      )
    }

    if (isDispatching) {
      throw new Error('Reducers may not dispatch actions.')
    }

    try {
      isDispatching = true
      currentState = currentReducer(currentState, action)
    } finally {
      isDispatching = false
    }

    const listeners = (currentListeners = nextListeners)
    for (let i = 0; i < listeners.length; i++) {
      const listener = listeners[i]
      listener()
    }

    return action
  }
```
The best explanation about it.
` Note that, if you use a custom middleware, it may wrap `dispatch()` to return something else (for example, a Promise you can await).`

**Comment by bloderxd** [comment]
Yeah, I agree with returning action in dispatch calls, I'm just wondering about what you said before "Actually the return of dispatch calls is state", because today our dispatch call doesn't have a return

**Comment by vitorprado** [comment]
@bloderxd sorry.. my mistake.. ignore it.

**Comment by vitorprado** [comment]
Implemented in #8
